src/bufr/BufrParser/Exports/Variables/AircraftAltitudeVariable.cpp

```diff
@@ -0,0 +1,339 @@
+                float pressureAltitudeRelativeToMeanSeaLevels = FLT_MAX;
+                float flightLevels = FLT_MAX;
+                float heights = FLT_MAX;
+                float heightOrAltitudes = FLT_MAX;
+                float flightLevelSTs = FLT_MAX;
+                if (!pressureQuery_.empty()) 
+                {
+                    pressures = static_cast<float>(map.at(getExportKey(ConfKeys::Pressure))->getAsFloat(idx));
+                    if (pressures != INT_MAX)
+                    {
+                        float ht_from_p = FLT_MAX;
+                        if (pressures < 22630)
+                        {
+                            ht_from_p = 11000 - (std::log1p(pressures/22630)/0.0001576106);
+                        } else {
+                            ht_from_p = (1.0-pow((pressures/101325),(1.0/5.256)))*(288.15/0.0065);
+                        }
+                        acftalt = ht_from_p;
+                    } else if (!aircraftIndicatedAltitudeQuery_.empty())
+                    {
+                        aircraftIndicatedAltitudes = static_cast<float>(map.at(getExportKey(ConfKeys::AircraftIndicatedAltitude))->getAsFloat(idx));
+                        if (aircraftIndicatedAltitudes != UINT_MAX)
+                        {
+                            acftalt = aircraftIndicatedAltitudes;
+                        }
+                    } else {
+                        acftalt = FLT_MAX;
+                    }
+                }
+                if (!pressureAltitudeRelativeToMeanSeaLevelQuery_.empty())
+                {
+                    pressureAltitudeRelativeToMeanSeaLevels = static_cast<float>(map.at(getExportKey(ConfKeys::PressureAltitudeRelativeToMeanSeaLevel))->getAsFloat(idx));
+                    if (pressureAltitudeRelativeToMeanSeaLevels != FLT_MAX)
+                    {
+                        acftalt = pressureAltitudeRelativeToMeanSeaLevels;
+                    }
+                }
+                if (!flightLevelQuery_.empty())
+                {
+                    flightLevels = static_cast<float>(map.at(getExportKey(ConfKeys::FlightLevel))->getAsFloat(idx));
+                    if (flightLevels != FLT_MAX)
+                    {
+                        acftalt = flightLevels;
+                    }
+                }
+
+                if (!heightQuery_.empty())
+                {
+                    heights = static_cast<float>(map.at(getExportKey(ConfKeys::Height))->getAsFloat(idx));
+                    if (heights != static_cast<float>(INT_MAX))
+                    {
+                        acftalt = heights;
+                    }
+                }
+
+                if (!heightOrAltitudeQuery_.empty())
+                {
+                    heightOrAltitudes = static_cast<float>(map.at(getExportKey(ConfKeys::HeightOrAltitude))->getAsFloat(idx));
+                    if (heightOrAltitudes != static_cast<float>(INT_MAX))
```

Comment:
I agree with @rmclaren's comments about comparing (for equality/non-equality) floats with other floats and with ints. A best practice could be to form the absolute value of the difference between the two numbers and check to see if that difference is less than a small tolerance value. This would help make the checking more portable across different compilers and platforms.